generators/Exercises/TwelveDays.cs

```diff
@@ -0,0 +1,50 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Generators.Input;
+using Generators.Output;
+using System.Linq;
+using Humanizer;
+
+namespace Generators.Exercises
+{
+    public class TwelveDays : GeneratorExercise
+    {
+        protected override void UpdateCanonicalData(CanonicalData canonicalData)
+        {
+            canonicalData.Exercise = "twelve-days-song";
+            foreach (var canonicalDataCase in canonicalData.Cases)
+            {
+                canonicalDataCase.UseVariableForExpected = true;
+                var properties = canonicalDataCase.Properties as Dictionary<string, object>;
+                var input = properties["input"] as Dictionary<string, object>;
+                if ((long)input["startVerse"] == (long)input["endVerse"])
```

Comment:
I understand you did this to keep the current implementation the same, but I think we should adhere to the canonical data. That means having just one property (`recite`) with start and end verse parameters. If we adhere to the canonical data, this customization can all go away.

Reply:
I understand, but it is in fact the other way around: we should be following the canonical data whenever possible.

Reply:
@ErikSchierboom OK, should I then remove

    public static string Sing()
    ...

from Example.cs and TwelveDays.cs ?

Reply:
Correct, that method should be remove in both files.